src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+function showMessage(message, className) {
+  const el = document.createElement('div');
+
+  el.setAttribute('data-qa', 'notification');
+  el.classList.add(className);
+  el.textContent = message;
+
+  document.body.appendChild(el);
+}
+
+let rightClick = false;
+let leftClick = false;
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      rightClick = true;
+    }
+
+    if (rightClick) {
+      resolve('Second promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    rightClick = true;
+
+    if (rightClick || leftClick) {
+      resolve('Second promise was resolved');
+    }
```

Comment:
The condition `if (rightClick || leftClick)` for resolving `secondPromise` might not be correct if the intention is to resolve only after a right click. Consider revising this logic to ensure it meets the intended behavior.